Assets/BossRoom/Scripts/Client/Game/Entity/ClientAvatarGuidHandler.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using BossRoom.Visual;
+using MLAPI;
+using UnityEngine;
+
+namespace BossRoom.Client
+{
+    /// <summary>
+    /// Client-side component that awaits a state change on an avatar's Guid, and fetches matching Avatar from the
+    /// AvatarRegistry, if possible. Once fetched, the Graphics GameObject is spawned.
+    /// </summary>
+    [RequireComponent(typeof(NetworkGuidState))]
+    public class ClientAvatarGuidHandler : NetworkBehaviour
+    {
+        [SerializeField]
+        ClientCharacter m_ClientCharacter;
+
+        [SerializeField]
+        CharacterClassContainer m_CharacterClassContainer;
+
+        [SerializeField]
+        NetworkGuidState m_NetworkGuidState;
+
+        [SerializeField]
+        AvatarRegistry m_AvatarRegistry;
+
+        void Awake()
```

Comment:
I would expect symmetry in this class
For awake and for register, I would expect a destroy and a unregister